### ISSUE TITLE
Tag specific atom feeds have incorrect "Next" and "Last" page links

### ISSUE THREAD

**Issue by lildude** [habari/habari#200]
Tag specific atom feeds accessed via http://example.com/tag/[tagname]/atom has incorrect "Next" and "Last" page links.

A simple illustration of the problem seen when viewing the atom feed for the "plugin" tag ( http://example.com/tag/plugin/atom )...

``` xml
<link rel="first" href="http://localhost/~col/habari/tag/plugin/atom/page/1?require_match%5B0%5D=Tag&amp;require_match%5B1%5D=rewrite_tag_exists" type="application/atom+xml" title="First Page"/>
<link rel="next" href="http://localhost/~col/habari/tag/plugin/atom/page/2?require_match%5B0%5D=Tag&amp;require_match%5B1%5D=rewrite_tag_exists" type="application/atom+xml" title="Next Page"/>
<link rel="last" href="http://localhost/~col/habari/tag/plugin/atom/page/43?require_match%5B0%5D=Tag&amp;require_match%5B1%5D=rewrite_tag_exists" type="application/atom+xml" title="Last Page"/>
```

In this case there are actually only 3 pages for the "plugin" tag, not 43.  There are however 43 "pages" for the entire site.

When viewing the atom feed for the 3rd page ( http://example.com/tag/plugin/atom/page/3 ) we get the following links:

``` xml
<link rel="first" href="http://localhost/~col/habari/tag/plugin/atom/page/1?require_match%5B0%5D=Tag&amp;require_match%5B1%5D=rewrite_tag_exists" type="application/atom+xml" title="First Page"/>
<link rel="previous" href="http://localhost/~col/habari/tag/plugin/atom/page/2?require_match%5B0%5D=Tag&amp;require_match%5B1%5D=rewrite_tag_exists" type="application/atom+xml" title="Previous Page"/>
<link rel="next" href="http://localhost/~col/habari/tag/plugin/atom/page/4?require_match%5B0%5D=Tag&amp;require_match%5B1%5D=rewrite_tag_exists" type="application/atom+xml" title="Next Page"/>
<link rel="last" href="http://localhost/~col/habari/tag/plugin/atom/page/43?require_match%5B0%5D=Tag&amp;require_match%5B1%5D=rewrite_tag_exists" type="application/atom+xml" title="Last Page"/>
```

There should be NO "Next Page" link here as there is no 4th page.

So in summary we have two issues here:

"Next" links should only be shown where there really is another page for that tag

... and ...

"Last" links should total the number of pages for that tag, not the whole site.
